common/SigUtil.cpp

```diff
@@ -283,10 +289,19 @@ namespace SigUtil
         }
     }
 
-    void setFatalSignals()
+    void setVersionInfo(const std::string &versionInfo)
+    {
+        if (VersionInfo)
+            free (VersionInfo);
+        VersionInfo = strdup(versionInfo.c_str());
```

Comment:
I wonder if ASAN or some similar tool will complain about the loss of this allocation. I suppose we can whitelist it.

Reply:
Possibly; but we can ignore that I think. Ultimately we could prolly use a std::string, and probably c_str() is signal safe - but, better to leak than to hide undefined behaviour in the code I think.